src/contract.rs

```diff
@@ -672,12 +677,12 @@ fn try_send<S: Storage, A: Api, Q: Querier>(
     amount: Uint128,
     msg: Option<Binary>,
 ) -> StdResult<HandleResponse> {
+    let sender = env.message.sender.clone();
     try_transfer_impl(deps, env, recipient, amount)?;
 
     let mut messages = vec![];
-    if let Some(msg) = msg {
-        try_add_receiver_api_callback(&mut messages, &deps.storage, recipient, msg)?;
-    }
+
+    try_add_receiver_api_callback(&mut messages, &deps.storage, recipient, msg, &sender, amount)?;
```

Comment:
If I don't do the clone() before the call to try_transfer_impl, the compiler complains about trying to use a field from env after it got moved during the try_transfer_impl call.  Is there another way around it or is cloning the sender ok?  I'm pretty new to rust (never saw a line of rust code before phase2 testnet started), so I'm certainly open to a better way.

Reply:
Ahhh, yeah i missed that point - I wasn't looking at the call to `try_transfer_impl`.
You can keep the `.clone()` here, but still change the signature of `try_add_receiver_api_callback`, and pass `sender` to it by value. That will save you the clone inside `try_add_receiver_api_callback`.

For someone that just got started, your code here looks really good! All my comments on this PR are pet peeves, since your code works perfectly as-is. These two comments on `.clone()` are just micro-optimzations because i know you can avoid the extra copies, which makes the function cheaper to execute. :)

Reply:
OK thanks!  Afraid I can't really take much credit for the code.  I've been looking at A LOT of example code in the last 6 or 7 weeks to get my footing and do a lot of copying lol.  Working on the updates now

Reply:
still, very cool :)